python/lsst/ts/standardscripts/maintel/laser_tracker/set_up.py

```diff
@@ -0,0 +1,97 @@
+# This file is part of ts_standardscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["SetUp"]
+
+import asyncio
+
+from lsst.ts import salobj
+from lsst.ts.idl.enum.LaserTracker import LaserStatus
+
+class SetUp(salobj.BaseScript):
+    """
+    Set up Laser Tracker.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+    **Checkpoints**
+
+    None
+
+    """
+
+    def __init__(self, index):
+        super().__init__(index, descr="Set up laser tracker.")
+
+        self.config = None
+
+        self.laser_tracker = salobj.RemoteGroup(
+            domain=self.domain, components=["LaserTracker:1"], log=self.log
+        )
+        self.timeout_laser_warmup = 60
+        self.timeout_laser_power = 30
+        self.timeout_std = 60
+
+    @classmethod
+    def get_schema(cls):
+        return None
+
+    async def configure(self, config):
+        self.config = config
+
+    async def startup(self):
+        await self.laser_tracker.start_task
+
+        await asyncio.gather(
+            self.laser_tracker.assert_liveliness(),
+            self.laser_tracker.assert_all_enabled()
+        )
+
+    async def check_laser_readiness(self):
+        # Check if the the self.laserTracker.laserStatus = ON or Warming
+
+        self.log.info("Checking if the laser is ready to be used.")
+
+        self.laser_tracker.rem.evt_laserStatus.flush()
+        laser_status = await self.laser_tracker.rem.evt_laserStatus.aget(
+            timeout=self.timeout_std,
+        )
+
+        while laser_status.status != LaserStatus.ON:
+            laser_status = await self.laser_tracker.rem.evt_laserStatus.next(
+                flush=False,
+                timeout=self.timeout_laser_warmup,
+            )
+
+    async def run(self):
+        await self.startup()
+        await self.laser_tracker.rem.cmd_laserPower.set_start(
```

Comment:
Consider adding a checkpoint before sending the command, maybe something like:

```
await self.checkpoint("Power up Laser Tracker.")
```

Reply:
Done